godot-core/src/docs.rs

```diff
@@ -23,40 +24,57 @@ pub struct StructDocs {
     pub members: &'static str,
 }
 
-/// Created for documentation on
+#[doc(hidden)]
+/// Keeps documentation for Inherent Implementation such as:
 /// ```ignore
 /// #[godot_api]
 /// impl Struct {
 ///     #[func]
 ///     /// This function panics!
 ///     fn panic() -> f32 { panic!() }
+///     #[signal]
+///     /// this signal signals
+///     fn documented_signal(p: Vector3, w: f64);
+///     #[constant]
+///     /// this constant consts
+///     const CON: i64 = 42;
+///
 /// }
 /// ```
 #[derive(Clone, Copy, Debug, Default)]
 pub struct InherentImplDocs {
-    pub methods: &'static str,
-    pub signals: &'static str,
-    pub constants: &'static str,
+    pub methods: Option<&'static str>,
+    pub signals_block: &'static str,
+    pub constants_block: &'static str,
 }
 
 #[derive(Default)]
 struct DocPieces {
     definition: StructDocs,
     inherent: InherentImplDocs,
-    virtual_methods: &'static str,
+    virtual_methods: Option<&'static str>,
 }
 
 #[doc(hidden)]
 /// This function scours the registered plugins to find their documentation pieces,
 /// and strings them together.
 ///
-/// It returns an iterator over XML documents.
+/// Returns an iterator over XML documents.
+///
+/// Documentation for signals and constants is being processed on compile time
+/// and can take a form of an already formatted xml `<block><doc></doc>…</block>` or an
+/// empty string if no such attribute has been documented.
+///
+/// Since documentation for methods comes from two different sources
+/// – Inherent Implementation (`methods`) and IGodotType trait implementation (`virtual_method_docs`) –
+/// it is undesirable to merge these on compile time, and they are being kept as an optional strings of not-yet-parented xml tags
+/// (or nothing if no method has been documented).
```

Comment:
```suggestion
/// Since documentation for methods comes from two different sources
/// -– inherent Implementation (`methods`) and I* trait implementation (`virtual_method_docs`) –-
/// it is undesirable to merge them at compile time. Instead, they are kept as optional strings of not-yet-parented XML tags
/// (or nothing if no method has been documented).
```

Reply:
Why a  `-– … –-` if I can ask?